src/hooks/use-list-behavior.js

```diff
@@ -0,0 +1,55 @@
+import { useCallback } from 'react'
+import safeInvoke from '../lib/safe-invoke'
+import { useRovingTabindex } from '../roving-tabindex'
+import { useClickable } from './use-clickable'
+
+export function useListBehavior({
+  disabled,
+  isMultiSelect,
+  isSelectable,
+  isSelected,
+  onSelect,
+  ref
+}) {
+  const { onClick: rovingTabIndexClick, onKeyDown: rovingTabIndexKeyDown, tabIndex: rovingTabIndex, ...rovingProps } = useRovingTabindex({
+    disabled,
+    isSelectable,
+    isSelected,
+    ref
+  })
+
+  const { onKeyDown: useClickableOnKeyDown, tabIndex: useClickableTabIndex } = useClickable({
+    disabled,
+    ref,
+    tabIndex: rovingTabIndex
+  })
```

Comment:
Use clickable accepts `onKeyDown` and will invoke it. That would simplify things a little here because you could remove `keyDownHandler`